be/src/storage/lake/compaction_policy.cpp

```diff
@@ -82,14 +82,30 @@ bool CompactionPolicy::is_real_time_compaction_strategy(const std::shared_ptr<co
     return metadata->has_compaction_strategy() && metadata->compaction_strategy() == CompactionStrategyPB::REAL_TIME;
 }
 
-StatusOr<uint32_t> primary_compaction_score_by_policy(TabletManager* tablet_mgr,
-                                                      const std::shared_ptr<const TabletMetadataPB>& metadata) {
+// In this function, we only count the number of filesets, not the number of SSTables.
```

Comment:
the number of sstables shall be worth taking into consideration even though it doesn't introduce the read amplification. However, compared to a number of small  sstables v.s. a few large sstables, the file IO efficiency should be better for the later.

The number of sstables shall be taking into consideration in a lower weight, so it could be still possible if a large number of small sorted sstable can be still chosen.

Not sure if my assumption is true here.